docs: Add flow versioning guide for NiFi 2 (#761)
stackabletech/nifi-operator: docs/modules/nifi/pages/usage_guide/flow-versioning.adoc

= Flow Versioning

IMPORTANT: The way described in this document to version flows is only available in NiFi 2.0.0 and later versions. Previous versions of NiFi relied exclusively on the NiFi Registry service for flow versioning.

Starting with Apache NiFi 2.0.0, you can connect directly to Git repositories using either GitHub or GitLab as your flow registry without requiring a separate NiFi Registry service. This document explains how to configure and use these registry clients to version your NiFi flows.

== Setting up the Registry Client

=== Configuration Steps

1. In your NiFi UI, click on the hamburger menu (☰) in the top right corner
2. Select "Controller Settings" > "Registry Clients"
3. Click the "+" icon to add a new registry client

> 4. Select "GitHubFlowRegistryClient" or "GitLabFlowRegistryClient" as the client type and click "Add"

Reviewer: I think quotes are fine for e.g. button references in the UI. This i would rather tick.
```suggestion
4. Select `GitHubFlowRegistryClient` or `GitLabFlowRegistryClient` as the client type and click "Add"
```

5. Click the three dots (⋮) on the right side of the client and select "Edit"
6. Fill in the following properties:

[cols="1,3"]
|===
|Property |Description

|API URL
|The API URL, like `https://api.github.com` for GitHub.com or a self-hosted GitLab URL

|Repository Owner
|The username or organization name that owns the repository

|Repository Name
|The name of the repository where flows will be stored

|Default Branch
|The default branch to use (e.g., `main`)

|Authentication Type
|The type of authentication to use. You can choose "None" for public repositories, "Personal Access Token" or "App Installation" are required for private repositories and are also helpful for public repositories in order to avoid rate limits.

|Repository Path
|Directory within the repository to store flows (e.g., `/flows`)
|===

Finally, click "Apply" to save the registry client configuration.

== Using Flow Registry Clients

> Once you've configured a Flow Registry Client, you can use it to version your Process Groups:

Reviewer: I would go for this or all lower case and spaced.
```suggestion
Once you've configured a "Git*FlowRegistryClient", you can use it to version your Process Groups:
```

Author: I went with the "all lower cased and spaced" variant, also in the heading of this section

1. Right-click on a Process Group in your NiFi canvas
2. Select "Version" > "Start version control"
3. Select your configured registry client, the branch to use, and the bucket (directory) to store flows in

> 4. Provide a Flow Name and optional comments

Reviewer: Unless you refer to the field, but then i would quote it?
```suggestion
4. Provide a flow name and optional comments
```
